crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -1211,6 +1212,35 @@ fn handle_import_from_comment<'a>(
     }
 }
 
+/// Attach an enclosed end-of-line comment to a [`ast::StmtWith`].
+///
+/// For example, given:
+/// ```python
+/// with ( # foo
+///     CtxManager1() as example1,
+///     CtxManager2() as example2,
+///     CtxManager3() as example3,
+/// ):
+///     ...
+/// ```
+///
+/// The comment will be attached to the [`ast::StmtWith`] node as a dangling comment, to ensure
+/// that it remains on the same line as the [`ast::StmtWith`] itself.
+fn handle_with_comment<'a>(
+    comment: DecoratedComment<'a>,
+    with_: &'a ast::StmtWith,
+) -> CommentPlacement<'a> {
+    if comment.line_position().is_end_of_line()
+        && with_.items.first().is_some_and(|with_item| {
+            with_.start() < comment.start() && comment.start() < with_item.start()
+        })
+    {
+        CommentPlacement::dangling(comment.enclosing_node(), comment)
+    } else {
+        CommentPlacement::Default(comment)
+    }
+}
```

Comment:
We could generalize this (the method is nearly identical to the `ImportFrom` example), but in my opinion it's fine to just duplicate it.